tests/e2e/tests/test_network.py

```diff
@@ -0,0 +1,131 @@
+#
+# Copyright 2023 Canonical, Ltd.
+#
+import json
+import logging
+from pathlib import Path
+
+import pytest
+from e2e_util import config, harness, util
+from e2e_util.config import MANIFESTS_DIR
+
+LOG = logging.getLogger(__name__)
+
+
+def test_network(h: harness.Harness, tmp_path: Path):
+    if not config.SNAP:
+        pytest.fail("Set TEST_SNAP to the path where the snap is")
+
+    snap_path = (tmp_path / "k8s.snap").as_posix()
+
+    LOG.info("Create instance")
+    instance_id = h.new_instance()
+
+    util.setup_k8s_snap(h, instance_id, snap_path)
+    h.exec(instance_id, ["k8s", "init"])
+    util.setup_network(h, instance_id)
+
+    p = h.exec(
+        instance_id,
+        [
+            "kubectl",
+            "--kubeconfig",
+            "/var/snap/k8s/common/etc/kubernetes/admin.conf",
+            "get",
+            "po",
+            "-n",
+            "kube-system",
+            "-l",
+            "k8s-app=cilium",
+            "-o",
+            "json",
+        ],
+        capture_output=True,
+    )
+
+    out = json.loads(p.stdout.decode())
+    assert len(out["items"]) > 0
+
+    cilium_pod = out["items"][0]
+
+    p = h.exec(
+        instance_id,
+        [
+            "kubectl",
+            "--kubeconfig",
+            "/var/snap/k8s/common/etc/kubernetes/admin.conf",
+            "exec",
+            "-it",
+            cilium_pod["metadata"]["name"],
+            "-n",
+            "kube-system",
+            "-c",
+            "cilium-agent",
+            "--",
+            "cilium",
+            "status",
+            "--brief",
+        ],
+        capture_output=True,
+    )
+
+    assert p.stdout.decode().strip() == "OK"
```

Comment:
@neoaggelos WDYT about changing the default behavior of `harness.exec` to return the decoded stdout? In most of the tests that we only want to check the output. 
We could still have something like a `decode=False` parameter in case we need to inspect the Command.